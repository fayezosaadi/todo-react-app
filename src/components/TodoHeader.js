import React from 'react';

const TodoHeader = ({ hide, hideCompleted }) => {
  return (
    <div className="todoListHeader">
      <h3>Todo List (1)</h3>
        <form>
          <label className="hide">
            <input name="hide" type="checkbox" checked={hide} onChange={hideCompleted} />
            Hide Completed Tasks
          </label>
        </form>
    </div>
  );
};

export default TodoHeader;
