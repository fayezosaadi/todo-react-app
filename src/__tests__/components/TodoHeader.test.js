import React from "react";
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import TodoHeader from "../../components/TodoHeader";

describe('<TodoHeader />',  () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState')
  useStateSpy.mockImplementation((init) => [init, setState]);

  it('should render', () => {
    render(<TodoHeader />)
  });

  it('should render and match the snapshot', () => {
    const component = renderer.create(<TodoHeader />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match the title', () => {
    const { getByText } = render(<TodoHeader />);
    const text = getByText('Todo List (1)');

    expect(text).toBeInTheDocument();
  });
});
