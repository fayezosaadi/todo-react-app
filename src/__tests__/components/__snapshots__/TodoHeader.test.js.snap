// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoHeader /> should render and match the snapshot 1`] = `
<div
  className="todoListHeader"
>
  <h3>
    Todo List (1)
  </h3>
  <form>
    <label
      className="hide"
    >
      <input
        name="hide"
        type="checkbox"
      />
      Hide Completed Tasks
    </label>
  </form>
</div>
`;
